### ISSUE TITLE
Search: Arrow navigation broken

### ISSUE THREAD

**Issue by mercury233** [microsoft/vscode#115953]
Behavior now: You can undo changes to the search string, but can't undo after clearing search results
![GIF 2021-2-6 11-50-18](https://user-images.githubusercontent.com/13391795/107108246-77a4b780-682e-11eb-9735-b39fa686ec67.gif)

What I want: When I undo after clearing search results, restore the last search string, in case of I click the clear button by accident.

**Comment by JacksonKearl** [comment]
You can use arrow keys to navigate through the search history, does that help?

**Comment by mercury233** [comment]
@JacksonKearl It work but it navigate to the last 2nd search history, for example, in my gif it will navigate to `ab` instead of `abc`

**Comment by JacksonKearl** [comment]
There's sometimes weirdness here where UpArraow then DownArrow is needed to get to the last entry, does that work in your case?

**Comment by mercury233** [comment]
@JacksonKearl yeah it works as a workaround, thank you

**Comment by mercury233** [comment]
@JacksonKearl no, the last search string before clear can't be found by up and down arrows

**Comment by JacksonKearl** [comment]
![Uploading Recording 2022-04-08 at 14.19.41.gif…]()

**Comment by mercury233** [comment]
@JacksonKearl it is not fully fixed in my client
>Version: 1.66.1 (system setup)
Commit: 8dfae7a5cd50421d10cd99cb873990460525a898
Date: 2022-04-06T14:50:12.141Z (2 days ago)
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Windows_NT x64 6.1.7601

![GIF 2022-4-9 6-13-04](https://user-images.githubusercontent.com/13391795/162539893-ecc9e6e5-6a66-46f7-982f-860879e06de6.gif)

the last search string can't be found by Arrow keys

**Comment by andreamah** [comment]
I can see this bug if I repeatedly clear and use the history value
